jscomp/syntax/ast_util.ml

```diff
@@ -286,23 +286,32 @@ let handle_debugger loc payload =
   else Location.raise_errorf ~loc "bs.raw can only be applied to a string"
 
 
-let handle_raw loc payload = 
+let handle_raw ?(check_js_regex = false) loc payload = 
   begin match Ast_payload.as_string_exp payload with 
     | None ->
       Location.raise_errorf ~loc
         "bs.raw can only be applied to a string "
 
-    | Some exp -> 
-      let pexp_desc = 
+    | Some exp ->
+      let exp_rtn exp_par = 
+      let pexp_desc =
         Parsetree.Pexp_apply (
           Exp.ident {loc; 
                      txt = 
                        Ldot (Ast_literal.Lid.js_unsafe, 
                              Literals.js_pure_expr)},
-          ["",exp]
+          ["",exp_par]
         )
       in
-      { exp with pexp_desc }
+      { exp_par with pexp_desc } in
+      match (exp, check_js_regex) with 
```

Comment:
IIRC, `as_string_exp` already did the destructuring to string, maybe you can push the validation logic down there, so here you don't need pattern match twice